dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/datastore/FileDataStoreHandler.kt

```diff
@@ -0,0 +1,289 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.datastore
+
+import android.text.format.DateUtils
+import androidx.annotation.WorkerThread
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.core.internal.persistence.datastore.ext.toByteArray
+import com.datadog.android.core.internal.persistence.datastore.ext.toInt
+import com.datadog.android.core.internal.persistence.datastore.ext.toLong
+import com.datadog.android.core.internal.persistence.file.FileReaderWriter
+import com.datadog.android.core.internal.persistence.file.createNewFileSafe
+import com.datadog.android.core.internal.persistence.file.deleteSafe
+import com.datadog.android.core.internal.persistence.file.existsSafe
+import com.datadog.android.core.internal.persistence.file.mkdirsSafe
+import com.datadog.android.core.internal.persistence.tlvformat.FileTLVBlockReader
+import com.datadog.android.core.internal.persistence.tlvformat.TLVBlock
+import com.datadog.android.core.internal.persistence.tlvformat.TLVBlockType
+import com.datadog.android.core.persistence.Serializer
+import java.io.File
+import java.util.Locale
+
+@Suppress("TooManyFunctions")
+internal class FileDataStoreHandler(
+    private val sdkInstanceId: String,
+    private val storageDir: File,
+    private val internalLogger: InternalLogger,
+    private val fileReaderWriter: FileReaderWriter,
+    private val fileTLVBlockReader: FileTLVBlockReader,
+    private val dataStoreFileHelper: DataStoreFileHelper
+) : DataStoreHandler {
+    @WorkerThread
+    override fun <T : Any> write(
+        dataStoreFileName: String,
+        featureName: String,
```

Comment:
should we simply host this handler in `SdkFeature` instead of `CoreFeature`? this will eliminate the need to pass `featureName` in the method and instead we could pass it in the constructor. This makes more sense.